src/SyntaxStream.php

```diff
@@ -0,0 +1,17 @@
+<?php declare(strict_types=1);
+
+namespace Yay;
+
+class SyntaxStream extends TokenStream {
```

Comment:
About the name, sorry for the back and fourth, but `ParsedTokenStream` looks more adequate :)

```diff
-class SyntaxStream extends TokenStream {
+class ParsedTokenStream extends TokenStream {
```